### PR TITLE
Deposit vdf

### PR DESCRIPTION
**Pull request by tomt1664**
Proof of work via a verifiable delay function is now required to generate a new shared key. 

The VDF challenge (a random 32 byte number) is returned from deposit_init, and the solution (which requires a few seconds of computation) is then returned and verified as part of keygen_first.

### CODE REVIEW

File: server/Settings.toml [hunk 1]
```diff
@@ -1,10 +1,9 @@
 electrum_server = "127.0.0.1:60401"
-network = "testnet"
+network = "regtest"
```
Comment (lawlawlaw) [inline server/Settings.toml:2]: revert to testnet

**Review by lawlawlaw — CHANGES_REQUESTED** [review]
- Remove the .DS_Store files
- There was a compile error when compiling the tests:
error[E0593]: closure is expected to take 4 arguments, but it takes 3 arguments
   --> integration-tests/src/test.rs:631:14
    |
631 |             .returning(|_user_id, _auth, _proof_key| Ok(()));
    |              ^^^^^^^^^ ----------------------------- takes 3 arguments
    |              |
    |              expected closure that takes 4 arguments

**Review by lawlawlaw — APPROVED** [review]

**Review by lawlawlaw — CHANGES_REQUESTED** [review]
.DSStore files should be removed.

**Comment by tomt1664** [comment]
VDF replaced with sha3 PoW. Challenge is now a 16 byte hex string, and solution a variable length hex string.

**Pull request merged**